packages/provider-elasticsearch/src/example-types/results/highlighting/response.js

```diff
@@ -0,0 +1,189 @@
+
+/*
+ * Group highlight results by their multifield. For example `city` and
+ * `city.subfield` will be grouped under `city`.
+ */
+let groupByMultiField = _.curry((schema, highlight) =>
+  groupByIndexed((v, path) => {
+    let [multi, sub] = path.split(lastWordRegex)
+    return schema.fields[multi]?.elasticsearch?.mapping?.fields?.[sub]
+      ? multi
+      : path
+  }, highlight)
+)
+
+/**
+ * Mutate hit `highlight`:
+ * 1. Fragments for text fields get merged.
+ * 2. Fragments for large (blob) text fields get concatenated.
+ * 3. Fragments for arrays get ordered based on the source array
+ */
+export let transformResponseHighlight = (
+  schema,
+  hit,
+  tags,
+  nestedArrayIncludes = {}
+) => {
+  let arrayOfObjectsPaths = _.keys(getArrayOfObjectsPathsMap(schema))
+
+  let getIndexedArrayObject = (arr, fragments, arrayPath, itemPath) => {
+    let fragmentsMap = _.groupBy(stripTags(tags), fragments)
+    return F.reduceIndexed(
+      (acc, item, index) => {
+        let fragments = fragmentsMap[itemPath ? _.get(itemPath, item) : item]
+        if (fragments) {
+          F.setOn(
+            F.dotJoin([`${index}`, itemPath]),
+            mergeHighlights(tags, ...fragments),
+            acc
+          )
+          for (let itemPath of nestedArrayIncludes[arrayPath] ?? []) {
+            F.updateOn(
+              F.dotJoin([`${index}`, itemPath]),
+              (highlight) => highlight ?? _.get(itemPath, item),
+              acc
+            )
+          }
+        }
+        return acc
+      },
+      arr,
+      _.get(arrayPath, hit._source)
+    )
+  }
+
+  let getArrayPath = (path) =>
+    isArrayOfScalarsField(schema.fields[path])
+      ? path
+      : findByPrefix(path, arrayOfObjectsPaths)
+
+  let highlight = _.flow(
+    groupByMultiField(schema),
+    _.mapValues(_.flatten),
+    F.reduceIndexed((acc, fragments, path) => {
+      let arrayPath = getArrayPath(path)
+      if (arrayPath) {
+        acc[arrayPath] = getIndexedArrayObject(
+          acc[arrayPath] ?? {},
+          fragments,
+          arrayPath,
+          path.slice(arrayPath.length + 1)
+        )
+      } else if (isBlobField(schema.fields[path])) {
+        acc[path] = fragments
+      } else {
+        acc[path] = mergeHighlights(tags, ...fragments)
+      }
+      return acc
+    }, {})
+  )(hit.highlight)
+
+  if (!_.isEmpty(highlight)) hit.highlight = highlight
+}
+
+/**
+ * Remove each path in `paths` from `hit._source`.
+ */
+export let removePathsFromSource = (schema, hit, paths) => {
+  // Nothing to do
+  if (_.isEmpty(paths)) return
+
+  // "aoo" stands for "array of objects", because I was tired of typing it out
+  // over and over again.
+  let aooMap = getArrayOfObjectsPathsMap(schema)
+  let allAooPaths = _.keys(aooMap)
+  let getAooPath = (path) => findByPrefix(path, allAooPaths)
+  let [aooPaths, otherPaths] = _.partition(getAooPath, paths)
+
+  let toRemove = {
+    ...F.arrayToObject(_.identity, _.constant(true), otherPaths),
```

Comment:
this is [`F.flags`](https://smartprocure.github.io/futil-js/#flags)

Reply:
`F.flags` unflattens the paths: `F.flags(['a.b']) => { a: { b: true } }` whereas I want `{ a.b: true }`